### ISSUE TITLE
Training on 640, Detecting on 4K.

### ISSUE THREAD

**Issue by NOxDriver** [ultralytics/yolov5#5100]
## ❔Question

I'm training an object detection model that can recognize animals in the wild in real-time. 

The idea is to have a model help you spot some wildlife. But, animals can be both far off and close up, so, when you resize an image to 640 pixel, the distant animals are not recognized by the model:
![image](https://user-images.githubusercontent.com/52410945/136557494-0bcc2ff5-b252-49d9-bbc7-e8dc807d9359.png)

Running detection with the same model but changing 640 to 4K images allow for the distant animals to be recognized too, but not the close animals.
![image](https://user-images.githubusercontent.com/52410945/136557696-2d1acf09-e2e7-4ce0-8798-6706833c946f.png)

But, it's impossible to train on 4K images due to the computing power needed.

I decided to try tiling our HD images so that we can train on 640 size, but detect at 4K. The issue I'm having now is animals close up in 4K have 20 640x640 size boxes drawn instead of one. 
![image](https://user-images.githubusercontent.com/52410945/136558256-ac872df4-8795-4747-b745-6a75cc0f8e21.png)

![image](https://user-images.githubusercontent.com/52410945/136562464-995b2212-a13b-490b-8a4d-804eb24e0fb8.png)

Also, when tiling the 4K images around the bounding box, sometimes the background in a label is now on a 640 block on its own. So it's technically still labeled as a lion, but when it's tiled, the lion isn't there, but the label remains. So in training, the background is now being taught as a lion. Is this just something we have to manually delete? Or would drawing an outline of a lion as opposed to a square make things better?
![For-github](https://user-images.githubusercontent.com/52410945/136562892-ee80c262-5c6f-477b-acce-c3a2158ee262.png)

Am I going about this the wrong way completely? Is there another simple solution? 

In summary, an animal can be far/small and close/big. How do you train on small images but then detect on 4K in a way that can pick up both scenarios?

**Comment by glenn-jocher** [comment]
@NOxDriver you want your training and deployment space to align for best real-world results. 

CNN's are translational invariant but not scale or rotation invariant. This means that the model will only be able to detect objects at sizes (in pixels) it has seen before during training. The size of the images is irrelevant, only the size of the objects matter.

If you want to detect an object that is 4k pixels then you must train on objects that are 4k pixels, and if you want to detect objects at 10 pixels then you must also train with objects at 10 pixels. If you train at 640 then this means the model has never seen (and will never predict) an object larger than 640 pixels.

For best results I would recommend training a P6 model at 1280 and then running inference at the same size, i.e.:
```
python train.py --img 1280 --weights yolov5m6.pt
python detect.py --img 1280
```

<img width="2029" alt="dataset" src="https://user-images.githubusercontent.com/26833433/136602414-b8ccffef-dc25-41aa-bdc5-ca3aa19171bb.png">
